packages/mds-stream/types.ts

```diff
@@ -29,3 +31,10 @@ export type StreamItemType = string
 export type StreamItemData = string
 export type StreamItem = [StreamItemID, [StreamItemType, StreamItemData]]
 export type ReadStreamResult = [Stream, StreamItem[]]
+
+export interface BadDataError {
+  recorded: Timestamp
+  provider_id: UUID
+  error_message: string
+  data: Partial<VehicleEvent>
+}
```

Comment:
Event timestamp may also be useful to have here as an optional field. We may also want to more strictly type the `error_message` field at some point with an enum of expected errors.

Reply:
I'd rather leave "optional event timestamp" as something that might exist in the `data` element, rather than pre-parse it, be responsible for its shape/format.